toolchain/semantics/semantics_handle_function.cpp

```diff
@@ -58,6 +58,11 @@ auto SemanticsHandleFunctionDefinitionStart(SemanticsContext& context,
 
   context.node_block_stack().Push();
   context.PushScope();
+  for (auto ref_id : context.semantics().GetNodeBlock(param_refs_id)) {
+    auto ref = context.semantics().GetNode(ref_id);
+    auto [name_id, target_id] = ref.GetAsBindName();
+    context.AddNameToLookupIgnoreConflicts(name_id, target_id);
+  }
```

Comment:
I feel like this would really benefit from a comment explaining why this is the right approach. Some of this is in the commit message, but I'd suggest including it here and being as specific as you can because this was pretty surprising to me.

For example, I *think* that the reason this makes sense is because the parameter names were already handled when we built the semantics IR for them, we just need te quickly put them back into name lookup here because they were kept out of name lookup while we finished the parameter block (to avoid default initializers etc from referencing other parameters). If I've understood this, yay, but would be good to document. If I've misunderstood it, then the correct explanation would be good. =D